src/theory/strings/term_registry.h

```diff
@@ -339,6 +337,8 @@ class TermRegistry : protected EnvObj
   TrustNode getRegisterTermAtomicLemma(Node n,
                                        LengthStatus s,
                                        std::map<Node, bool>& reqPhase);
+  /** register term n, called when it is known n is not already registered */
```

Comment:
is it possible to assert this at the beginning of the implementation of this function?